src/common/dns_utils.cpp

```diff
@@ -447,19 +447,28 @@ bool load_txt_records_from_dns(std::vector<std::string> &good_records, const std
   std::uniform_int_distribution<int> dis(0, dns_urls.size() - 1);
   size_t first_index = dis(gen);
 
-  bool avail, valid;
+  // send all requests in parallel
+  std::vector<boost::thread> threads(dns_urls.size());
+  std::deque<bool> avail(dns_urls.size(), false), valid(dns_urls.size(), false);
+  for (size_t n = 0; n < dns_urls.size(); ++n)
+  {
+    threads[n] = boost::thread([n, dns_urls, &records, &avail, &valid](){
```

Comment:
Was this supposed to be `&dns_urls` ? Also, should boost::thread have the stack size attribute set? Perhaps we need to create a global function somewhere that set this automatically?

```c++
template<typename... T>
boost::thread start_thread(T&&... args) {
    return boost::thread( ... attributes ..., std::forward<T>(args)...);
}
```

Reply:
They're short lived "just wait for a reply" threads, so don't need a large stack. I did not add & because I do not need to modify the parent copy. I'm not sure how to pass a const ref instead of a ref though. Do you know ?

Reply:
The only method I've found is to alias it locally and then capture by ref:
```c++
const auto& cfoo = foo;
[&cfoo] () { .... };
```
which kind of stinks.